vm/src/exceptions.rs

```diff
@@ -796,7 +796,9 @@ impl ExceptionZoo {
             // TODO: members
             "filename" => ctx.none(),
             "lineno" => ctx.none(),
+            "end_lineno" => ctx.none(),
             "offset" => ctx.none(),
+            "end_offset" => ctx.none(),
```

Comment:
Thank you for your help in solving the problem. 
In fact, it was not confirmed whether the test file was runable because it was a pull request only until the commit where the test file was just copied. 
At The local machine, I add the expectedFailure tag and skip tag to make test runable. 

The biggest problem is that the test_no_hang_on_context_chain_cycle1,2,3 test in Lib/test/test_exception.py is stuck in an infinite loop, so the test suite have not ended. 
(Still don't know why they are stuck)

I'm reviewing the entire test to check if there are any other problems, so I'll leave a comment again when the results come out. 

I didn't know that the order of attributes could be a problem. Could you explain a little bit more about the mechanism that could be a problem?

Reply:
> I didn't know that the order of attributes could be a problem. Could you explain a little bit more about the mechanism that could be a problem?

My mistake. I was caught off guard by the Unicode-related errors, as they have similarly-named attributes.

___

I'd recommend taking a look at the definition of `fn new_syntax_error()` (in `vm/src/vm/vm_new.rs`) as well as where that method is invoked. (You may need to reach into RustPython's `compiler` crate to implement those new attributes properly.)

Reply:
the easiest way to diagnosing hanging problem is checking callstack when it hanged. I usually use debugger but there will be other ways too.
tough you may not need it right now